tests/blackbox/heading-id.t

```diff
@@ -0,0 +1,8 @@
+  $ omd << "MD"
+  > ### This is an Header Without Id
+  > ### 1 2 Header that starts with 2 numbers
+  > ### Header with an id {#header-id}
+  > MD
+  <h3 id="this-is-an-header-without-id">This is an Header Without Id</h3>
+  <h3 id="header-that-starts-with-2-numbers">1 2 Header that starts with 2 numbers</h3>
+  <h3 id="header-id">Header with an id</h3>
```

Comment:
What's the reason for stripping of leading digits? I'm not an HTML expert, but looking at [the spec for HTML5](https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute), it looks like [id attributes can include almost any characters]?:

> There are no other restrictions on what form an ID can take; in particular, IDs can consist of just digits, start with a digit, start with an underscore, consist of just punctuation, etc.

We can probably expand this test to cover some other edge cases, like with some unusual characters (including unicode) mixed in?

Reply:
Thanks for the review @shonfeder 😊

> What's the reason for stripping of leading digits? I'm not an HTML expert, but looking at [the spec for HTML5](https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute), it looks like [id attributes can include almost any characters]?:

Because I noticed that's what pandoc does, and the original issue says: 

> also to make sure there is a bit of standardization (maybe choose same IDs as pandoc).

I'm not familiar with pandoc so please let me know if I misunderstood something, it can be easily changed.

> We can probably expand this test to cover some other edge cases, like with some unusual characters (including unicode) mixed in?

Makes sense, I'll try to come up with a few more test cases.